examples/vision/knowledge_distillation.py

```diff
@@ -0,0 +1,276 @@
+"""
+Title: Knowledge Distillation
+Author: [Kenneth Borup](https://twitter.com/Kennethborup)
+Date created: 2020/09/01
+Last modified: 2020/09/01
+Description: Implementation of classical Knowledge Distillation.
+"""
+
+"""
+## Introduction to Knowledge Distillation
+
+Knowledge Distillation (or student-teacher training) is a procedure for model
+compression, in which a small (student) model is trained to match a large pre-trained
+(teacher) model. Intrinsic knowledge is transferred from the teacher model to the student
+by minimizing a loss function, aimed at matching softened teacher logits as well as
+ground-truth labels.
+The logits are softened by applying a temperature in the softmax, effectively smoothing
+out the probability distribution and revealing inter-class relationships learned by the
+teacher.
+
+**Reference:**
+
+- [Hinton et al. (2015)](https://arxiv.org/abs/1503.02531)
+"""
+
+"""
+## Setup
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import numpy as np
+
+tf.random.set_seed(0)
+np.random.seed(0)
+
+"""
+## Construct `Distiller()` class
```

Comment:
Please aerate the text a bit: line break after titles, line break before bullet points list (important to ensure proper markdown rendering)